tests/run-make/translation/rmake.rs

```diff
@@ -0,0 +1,89 @@
+// of each test case.
+// See https://github.com/rust-lang/rust/pull/95512
+
+//@ needs-symlink
+
+use run_make_support::{path, rfs, rustc};
+
+fn main() {
+    // Check that the test works normally, using the built-in fallback bundle.
+    rustc().input("test.rs").run_fail().assert_stderr_contains("struct literal body without path");
+    // Check that a primary bundle can be loaded and will be preferentially used
+    // where possible.
+    rustc()
+        .arg("-Ztranslate-additional-ftl=working.ftl")
+        .input("test.rs")
+        .run_fail()
+        .assert_stderr_contains("this is a test message");
+    // Check that a primary bundle with a broken message (e.g. a interpolated
+    // variable is missing) will use the fallback bundle.
+    rustc()
+        .arg("-Ztranslate-additional-ftl=missing.ftl")
+        .input("test.rs")
+        .run_fail()
+        .assert_stderr_contains("struct literal body without path");
+    // Check that a primary bundle without the desired message will use the fallback
+    // bundle.
+    rustc()
+        .arg("-Ztranslate-additional-ftl=broken.ftl")
+        .input("test.rs")
+        .run_fail()
+        .assert_stderr_contains("struct literal body without path");
+
+    // Check that a locale can be loaded from the sysroot given a language
+    // identifier by making a local copy of the sysroot and adding the custom locale
+    // to it.
+    let sysroot = rustc().print("sysroot").run().stdout_utf8();
+    let sysroot = sysroot.trim();
+    rfs::create_dir("fakeroot");
+    symlink_all_entries(&sysroot, "fakeroot");
+    rfs::remove_file("fakeroot/lib");
+    rfs::create_dir("fakeroot/lib");
+    symlink_all_entries(path(&sysroot).join("lib"), "fakeroot/lib");
+    rfs::remove_file("fakeroot/lib/rustlib");
+    rfs::create_dir("fakeroot/lib/rustlib");
+    symlink_all_entries(path(&sysroot).join("lib/rustlib"), "fakeroot/lib/rustlib");
+    rfs::remove_file("fakeroot/lib/rustlib/src");
+    rfs::create_dir("fakeroot/lib/rustlib/src");
+    symlink_all_entries(path(&sysroot).join("lib/rustlib/src"), "fakeroot/lib/rustlib/src");
+    // When download-rustc is enabled, `sysroot` will have a share directory. Delete the link to it.
+    if path("fakeroot/share").exists() {
+        rfs::remove_file("fakeroot/share");
+    }
+    rfs::create_dir_all("fakeroot/share/locale/zh-CN");
+    rfs::create_symlink("working.ftl", "fakeroot/share/locale/zh-CN/basic-translation.ftl");
+    rustc()
+        .arg("-Ztranslate-lang=zh-CN")
+        .input("test.rs")
+        .sysroot("fakeroot")
+        .run_fail()
+        .assert_stderr_contains("this is a test message");
+
+    // Check that the compiler errors out when the sysroot requested cannot be
+    // found. This test might start failing if there actually exists a Klingon
+    // translation of rustc's error messages.
+    rustc()
+        .arg("-Ztranslate-lang=tlh")
+        // .input("test.rs")
```

Comment:
Remark: commented out

Reply:
Original test:

```make
sysroot-missing:
	$(RUSTC) $< -Ztranslate-lang=tlh 2>&1 | $(CGREP) "missing locale directory"
```

Note how the ususal `test.rs` next to the section header `sysroot-missing:` is absent. I have no idea what the `$<` is slotting in.